.github/workflows/nightly.yml

```diff
@@ -0,0 +1,67 @@
+
+          - runson: codebuild-test-${{ github.run_id }}-${{ github.run_attempt }} # linux arm
+            arch: aarch64
+    env:
+      # these need to be specific to the job so that CodeBuild can read them, so there is a bit of duplication
+      REPOSITORY: openorclose/async-profiler
+      LATEST_RELEASE_ID: 176384274
+      PAST_RELEASES_ID: 174369452
+    runs-on: ${{ matrix.runson }}
+    container:
+      image: ${{ matrix.image }}
+      options: --privileged # needed in order to use sysctl
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build, test, and publish
+        run: |
+          if [[ "${{ matrix.runson }}" != "macos-latest" ]]; then # configuration needed only for linux
+            sudo sysctl kernel.perf_event_paranoid=1 && sudo sysctl kernel.kptr_restrict=0 # allow tests to execute
+            CC=/usr/local/musl/${{ matrix.arch }}/bin/musl-gcc
+          fi
+          git config --global --add safe.directory $PWD # needed to avoid `fatal: detected dubious ownership in repository at '/__w/async-profiler/async-profiler'` error
+          make GITHUB_TOKEN=${{ github.token }} RELEASE_ID=${{ env.LATEST_RELEASE_ID }} COMMIT_TAG=true release test publish-github-artifact # build, test, and publish binaries to latest release
+          make GITHUB_TOKEN=${{ github.token }} RELEASE_ID=${{ env.PAST_RELEASES_ID }} COMMIT_TAG=true release publish-github-artifact # build and publish binaries to last 5 commits release
+  delete-past-releases:
+    needs: [build-test-publish]
+    env:
+      REPOSITORY: openorclose/async-profiler
+    strategy:
+      matrix:
+        include:
+          - release_id: 174369452 # cannot access env vars from strategy
+            commits_to_keep: 5
+          - release_id: 176384274
+            commits_to_keep: 1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 5
+      - name: Delete old commits
+        run: |
+          # get the last commits_to_keep commit hashes in the format 
+          #    contains($HASH1) or contains($HASH2) or
+          # to insert into jq
+          last_n_hashes=$(git log  --format='contains("%h") or' --abbrev=8 -n${{ matrix.commits_to_keep }} | tr '\n' ' ')
+          echo $last_n_hashes
+          # get the ids of all published artifacts in the nightly release that are not for the latest commits_to_keep hashes
+          ids=$(curl -L  -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" -H "X-GitHub-Api-Version: 2022-11-28"  "https://api.github.com/repos/${REPOSITORY}/releases/${{ matrix.release_id }}/assets" | jq ".[]|select(.name | ($last_n_hashes false) | not).id")
+          # delete all assets in the nightly release
+          for id in $ids; do curl -L -X DELETE -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${REPOSITORY}/releases/assets/${id}"; done
```

Comment:
All these delete stuff gets a bit complicated. Would be simpler if we were to delete artifacts basing on their age rather than count? wdyt?

Reply:
https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#artifact-and-log-retention-policy

Maybe instead of an archive release we can publish the artifacts to the workflow directly instead? And we can set the retention date to something like 30 days. The benefit is that we don't have to worry about managing them since they will be automatically deleted

Reply:
Yeah, I like the idea. This should makes things a lot cleaner.

Reply:
https://github.com/openorclose/async-profiler/actions/runs/11016317799

Implemented here, now people can access the latest binaries by going to the commit and clicking on the workflow run